### ISSUE TITLE
esp-idf latest build error in protocomm

### ISSUE THREAD

**Issue by buckybots** [espressif/esp-idf#2598]
switched to master branch from v3.1-rc2 and no longer compiles with make.

Here is the broken part :+1: 

C:/linuxShared/esp/esp-idf/components/protocomm/src/transports/protocomm_httpd.c:33:38: error: unknown type name 'httpd_req_t'
 static esp_err_t common_post_handler(httpd_req_t *req)
                                      ^
C:/linuxShared/esp/esp-idf/components/protocomm/src/transports/protocomm_httpd.c: In function 'protocomm_httpd_add_endpoint':
C:/linuxShared/esp/esp-idf/components/protocomm/src/transports/protocomm_httpd.c:138:30: error: implicit declaration of function 'strlen' [-Werror=implicit-function-declaration]
     char* ep_uri = calloc(1, strlen(ep_name) + 2);
                              ^
C:/linuxShared/esp/esp-idf/components/protocomm/src/transports/protocomm_httpd.c:138:30: warning: incompatible implicit declaration of built-in function 'strlen'
C:/linuxShared/esp/esp-idf/components/protocomm/src/transports/protocomm_httpd.c:138:30: note: include '<string.h>' or provide a declaration of 'strlen'
C:/linuxShared/esp/esp-idf/components/protocomm/src/transports/protocomm_httpd.c:146:5: error: unknown type name 'httpd_uri_t'
     httpd_uri_t config_handler = {
     ^
C:/linuxShared/esp/esp-idf/components/protocomm/src/transports/protocomm_httpd.c:147:9: error: field name not in record or union initializer
         .uri      = ep_uri,
         ^
C:/linuxShared/esp/esp-idf/components/protocomm/src/transports/protocomm_httpd.c:147:9: note: (near initialization for 'config_handler')
C:/linuxShared/esp/esp-idf/components/protocomm/src/transports/protocomm_httpd.c:147:21: warning: initialization makes integer from pointer without a cast [-Wint-conversion]
         .uri      = ep_uri,
                     ^
C:/linuxShared/esp/esp-idf/components/protocomm/src/transports/protocomm_httpd.c:147:21: note: (near initialization for 'config_handler')
C:/linuxShared/esp/esp-idf/components/protocomm/src/transports/protocomm_httpd.c:148:9: error: field name not in record or union initializer
         .method   = HTTP_POST,
         ^
C:/linuxShared/esp/esp-idf/components/protocomm/src/transports/protocomm_httpd.c:148:9: note: (near initialization for 'config_handler')
C:/linuxShared/esp/esp-idf/components/protocomm/src/transports/protocomm_httpd.c:148:21: warning: excess elements in scalar initializer
         .method   = HTTP_POST,
                     ^
C:/linuxShared/esp/esp-idf/components/protocomm/src/transports/protocomm_httpd.c:148:21: note: (near initialization for 'config_handler')
C:/linuxShared/esp/esp-idf/components/protocomm/src/transports/protocomm_httpd.c:149:9: error: field name not in record or union initializer
         .handler  = common_post_handler,
         ^
C:/linuxShared/esp/esp-idf/components/protocomm/src/transports/protocomm_httpd.c:149:9: note: (near initialization for 'config_handler')
C:/linuxShared/esp/esp-idf/components/protocomm/src/transports/protocomm_httpd.c:149:21: error: 'common_post_handler' undeclared (first use in this function)
         .handler  = common_post_handler,
                     ^
C:/linuxShared/esp/esp-idf/components/protocomm/src/transports/protocomm_httpd.c:149:21: note: each undeclared identifier is reported only once for each function it appears in
C:/linuxShared/esp/esp-idf/components/protocomm/src/transports/protocomm_httpd.c:149:21: warning: excess elements in scalar initializer
C:/linuxShared/esp/esp-idf/components/protocomm/src/transports/protocomm_httpd.c:149:21: note: (near initialization for 'config_handler')
C:/linuxShared/esp/esp-idf/components/protocomm/src/transports/protocomm_httpd.c:150:9: error: field name not in record or union initializer
         .user_ctx = NULL
         ^
C:/linuxShared/esp/esp-idf/components/protocomm/src/transports/protocomm_httpd.c:150:9: note: (near initialization for 'config_handler')
C:/linuxShared/esp/esp-idf/components/protocomm/src/transports/protocomm_httpd.c:155:5: error: unknown type name 'httpd_handle_t'
     httpd_handle_t *server = (httpd_handle_t *) pc_httpd->priv;
     ^
C:/linuxShared/esp/esp-idf/components/protocomm/src/transports/protocomm_httpd.c:155:31: error: 'httpd_handle_t' undeclared (first use in this function)
     httpd_handle_t *server = (httpd_handle_t *) pc_httpd->priv;
                               ^
C:/linuxShared/esp/esp-idf/components/protocomm/src/transports/protocomm_httpd.c:155:47: error: expected expression before ')' token
     httpd_handle_t *server = (httpd_handle_t *) pc_httpd->priv;
                                               ^
C:/linuxShared/esp/esp-idf/components/protocomm/src/transports/protocomm_httpd.c:156:16: error: implicit declaration of function 'httpd_register_uri_handler' [-Werror=implicit-function-declaration]
     if ((err = httpd_register_uri_handler(*server, &config_handler)) != ESP_OK) {
                ^
C:/linuxShared/esp/esp-idf/components/protocomm/src/transports/protocomm_httpd.c: In function 'protocomm_httpd_remove_endpoint':
C:/linuxShared/esp/esp-idf/components/protocomm/src/transports/protocomm_httpd.c:175:30: warning: incompatible implicit declaration of built-in function 'strlen'
     char* ep_uri = calloc(1, strlen(ep_name) + 2);
                              ^
C:/linuxShared/esp/esp-idf/components/protocomm/src/transports/protocomm_httpd.c:175:30: note: include '<string.h>' or provide a declaration of 'strlen'
C:/linuxShared/esp/esp-idf/components/protocomm/src/transports/protocomm_httpd.c:184:5: error: unknown type name 'httpd_handle_t'
     httpd_handle_t *server = (httpd_handle_t *) pc_httpd->priv;
     ^
C:/linuxShared/esp/esp-idf/components/protocomm/src/transports/protocomm_httpd.c:184:31: error: 'httpd_handle_t' undeclared (first use in this function)
     httpd_handle_t *server = (httpd_handle_t *) pc_httpd->priv;
                               ^
C:/linuxShared/esp/esp-idf/components/protocomm/src/transports/protocomm_httpd.c:184:47: error: expected expression before ')' token
     httpd_handle_t *server = (httpd_handle_t *) pc_httpd->priv;
                                               ^
C:/linuxShared/esp/esp-idf/components/protocomm/src/transports/protocomm_httpd.c:185:16: error: implicit declaration of function 'httpd_unregister_uri_handler' [-Werror=implicit-function-declaration]
     if ((err = httpd_unregister_uri_handler(*server, ep_uri, HTTP_POST)) != ESP_OK) {
                ^
C:/linuxShared/esp/esp-idf/components/protocomm/src/transports/protocomm_httpd.c: In function 'protocomm_httpd_start':
C:/linuxShared/esp/esp-idf/components/protocomm/src/transports/protocomm_httpd.c:211:33: error: 'httpd_handle_t' undeclared (first use in this function)
     pc->priv = calloc(1, sizeof(httpd_handle_t));
                                 ^
C:/linuxShared/esp/esp-idf/components/protocomm/src/transports/protocomm_httpd.c:218:5: error: unknown type name 'httpd_config_t'
     httpd_config_t server_config   = HTTPD_DEFAULT_CONFIG();
     ^
C:/linuxShared/esp/esp-idf/components/protocomm/src/transports/protocomm_httpd.c:218:38: error: implicit declaration of function 'HTTPD_DEFAULT_CONFIG' [-Werror=implicit-function-declaration]
     httpd_config_t server_config   = HTTPD_DEFAULT_CONFIG();
                                      ^
C:/linuxShared/esp/esp-idf/components/protocomm/src/transports/protocomm_httpd.c:219:18: error: request for member 'server_port' in something not a structure or union
     server_config.server_port      = config->port;
                  ^
C:/linuxShared/esp/esp-idf/components/protocomm/src/transports/protocomm_httpd.c:220:18: error: request for member 'stack_size' in something not a structure or union
     server_config.stack_size       = config->stack_size;
                  ^
C:/linuxShared/esp/esp-idf/components/protocomm/src/transports/protocomm_httpd.c:221:18: error: request for member 'task_priority' in something not a structure or union
     server_config.task_priority    = config->task_priority;
                  ^
C:/linuxShared/esp/esp-idf/components/protocomm/src/transports/protocomm_httpd.c:222:18: error: request for member 'lru_purge_enable' in something not a structure or union
     server_config.lru_purge_enable = true;
                  ^
C:/linuxShared/esp/esp-idf/components/protocomm/src/transports/protocomm_httpd.c:223:18: error: request for member 'max_open_sockets' in something not a structure or union
     server_config.max_open_sockets = 1;
                  ^
C:/linuxShared/esp/esp-idf/components/protocomm/src/transports/protocomm_httpd.c:226:16: error: implicit declaration of function 'httpd_start' [-Werror=implicit-function-declaration]
     if ((err = httpd_start((httpd_handle_t *)pc->priv, &server_config)) != ESP_OK) {
                ^
C:/linuxShared/esp/esp-idf/components/protocomm/src/transports/protocomm_httpd.c:226:45: error: expected expression before ')' token
     if ((err = httpd_start((httpd_handle_t *)pc->priv, &server_config)) != ESP_OK) {
                                             ^
C:/linuxShared/esp/esp-idf/components/protocomm/src/transports/protocomm_httpd.c:218:20: warning: variable 'server_config' set but not used [-Wunused-but-set-variable]
     httpd_config_t server_config   = HTTPD_DEFAULT_CONFIG();
                    ^
C:/linuxShared/esp/esp-idf/components/protocomm/src/transports/protocomm_httpd.c: In function 'protocomm_httpd_stop':
C:/linuxShared/esp/esp-idf/components/protocomm/src/transports/protocomm_httpd.c:241:9: error: unknown type name 'httpd_handle_t'
         httpd_handle_t *server_handle = (httpd_handle_t *) pc_httpd->priv;
         ^
C:/linuxShared/esp/esp-idf/components/protocomm/src/transports/protocomm_httpd.c:241:42: error: 'httpd_handle_t' undeclared (first use in this function)
         httpd_handle_t *server_handle = (httpd_handle_t *) pc_httpd->priv;
                                          ^
C:/linuxShared/esp/esp-idf/components/protocomm/src/transports/protocomm_httpd.c:241:58: error: expected expression before ')' token
         httpd_handle_t *server_handle = (httpd_handle_t *) pc_httpd->priv;
                                                          ^
C:/linuxShared/esp/esp-idf/components/protocomm/src/transports/protocomm_httpd.c:242:9: error: implicit declaration of function 'httpd_stop' [-Werror=implicit-function-declaration]
         httpd_stop(*server_handle);
         ^
C:/linuxShared/esp/esp-idf/components/protocomm/src/transports/protocomm_httpd.c: At top level:
C:/linuxShared/esp/esp-idf/components/protocomm/src/transports/protocomm_httpd.c:29:17: warning: 'session_id' defined but not used [-Wunused-variable]
 static uint32_t session_id = PROTOCOMM_NO_SESSION_ID;
                 ^
cc1.exe: some warnings being treated as errors
make[1]: *** [/c/linuxShared/esp/esp-idf/make/component_wrapper.mk:286: src/transports/protocomm_httpd.o] Error 1
make: *** [C:\linuxShared\esp\esp-idf/make/project.mk:505: component-protocomm-build] Error 2

**Comment by igrr** [comment]
Possibly same issue as https://github.com/espressif/esp-idf/issues/2583#issuecomment-431292166?

**Comment by buckybots** [comment]
Yes, same exact issue, name conflict with http-server.h.  Was using your quick and dirty http-server, and now that a http-server is part of esp-idf  this component is no longer needed.  Easy fix, just removed my http-server component.
